tests/pyccel/scripts/decorators_elemental.py

```diff
@@ -0,0 +1,18 @@
+from pyccel.decorators import types, elemental
+from numpy import array
+from numpy import zeros_like
+
+@elemental
+@types(float)
+def square(x):
+    s = x*x
+    return s
+
+a = 2.0
+b = square(a)
+print(b)
+
+xs = array([1., 2., 3.])
+ys = zeros_like(xs)
+ys = square(xs)
```

Comment:
It appears that Pyccel can only process a function with an array result if it's elemental. But then we should not need to manually call `zeros_like` in order to allocate the array, Pyccel should be able to do it automatically. This is related to issue #359

Reply:
I don't think we could return the result of an elemental call currently with epyccel, but I have made the modifications necessary to allow the allocation when it is called from fortran code

Reply:
The allocation is done in the same place for both `a=b+c` and `ys = square(xs)` so they should be fixed in the same way when #359 is fixed

Reply:
Perfect, that's all we need for now! Thanks